cylc/flow/option_parsers.py

```diff
@@ -583,3 +678,110 @@ def __call__(self, **kwargs) -> Values:
             setattr(opts, key, value)
 
         return opts
+
+
+def appendif(list_, item):
+    """Avoid duplicating items in output list"""
+    if item not in list_:
+        list_.append(item)
+    return list_
+
+
+def combine_options_pair(first_list, second_list):
+    """Combine two option lists.
+
+    Scenarios:
+        - Arguments are identical - return this argument.
+        - Arguments are not identical but have some common label strings,
+          i.e. both arguments can be invoked using `-f`.
+          - If there are non-shared label strings strip the shared ones.
+          - Otherwise raise an error.
+          E.g: If `command-A` has an option `-f` or `--file` and
+          `command-B has an option `-f` or `--fortran`` then
+          `command-A+B` will have options `--fortran` and `--file` but _not_
+          `-f`, which would be confusing.
+
+    TODO: Record which list each option came from, so that you can tell
+    the user which component script is being referred to by each option.
+    """
+
+    output = []
+
+    for first, second in product(first_list, second_list):
+        if first == second and first not in output:
+            output = appendif(output, first)
+
+        # If any of the argument name identical:
+        elif (
+            any([i[0] == i[1] for i in product(second[ARGS], first[ARGS])])
+            and first != second
```

Comment:
Put the fast check before the slow check and avoid unnecessary iteration:

```suggestion
            first != second
            and any(i[0] == i[1] for i in product(second[ARGS], first[ARGS]))
```